### PR TITLE
Nodes Recovery::Update nodesource addons to scheduling changes

### PR DESCRIPTION
**Pull request by jrochas**

### CODE REVIEW

File: infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                nodesPerInstances.get(instanceId).remove(nodeName);
+                logger.info("Removed node : " + nodeName);
+                if (nodesPerInstances.get(instanceId).isEmpty()) {
+                    connectorIaasController.terminateInstance(infrastructureId, instanceId);
+                    nodesPerInstances.remove(instanceId);
+                    logger.info("Removed instance : " + instanceId);
+                }
+                // finally write to the runtime variable map
+                runtimeVariables.put(NODES_PER_INSTANCES_KEY, nodesPerInstances);
+                return null;
```
Comment (MaelAudren) [inline infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AbstractAddonInfrastructure.java]: return null in a void method ? I think you can just delete it.

**Comment by jrochas** [comment]
I agree that more code should be factorized in the super class. Thanks.

**Comment by vinseon** [comment]
What about failed checks? Is this PR linked to another one (scheduling?)

**Comment by jrochas** [comment]
I assume the build fails because the PR take the code of scheduling that is available in the artifact repository, which reflects the master branch of scheduling, whereas it requires the code that is on the feature/nodes-recovery branch of scheduling.

**Comment by vinseon** [comment]
OK so this should be merged only after `feature/nodes-recovery` branch of scheduling is merged.

**Pull request merged**